base/abstractarraymath.jl

```diff
@@ -2,8 +2,6 @@
 
  ## Basic functions ##
 
-isinteger(x::AbstractArray) = all(isinteger,x)
-isinteger{T<:Integer,n}(x::AbstractArray{T,n}) = true
```

Comment:
With this change, we will lose this optimization. It seems to be blocked by the bounds check in `all`; adding `@inbounds` to `all` allows the loop to be optimized away.